sdmx30-infomodel/src/main/java/com/epam/jsdmx/infomodel/sdmx30/SdmxUrn.java

```diff
@@ -98,10 +98,38 @@ public static String parseToShortItemUrnString(String fullUrn) {
         return toShortItemUrnString(components.getAgency(), components.getId(), components.getVersion(), components.getItemId());
     }
 
+    /**
+     * @deprecated due to reporting incorrect structure class for items fo item schemes and (m)dsd components.
+     * Use {@link #getItemUrnString(ArtefactReference, IdentifiableArtefact)}
+     */
+    @Deprecated
     public static String getItemUrnString(String containerUrn, String containedId) {
         return containerUrn + "." + containedId;
     }
 
+    public static String getItemUrnString(ArtefactReference container, IdentifiableArtefact contained) {
+        if (container == null) {
```

Comment:
nit: Objects.requireNonNull could be used instead

Reply:
indeed, but the goal is to avoid NPE, and report `IllegalArguemntException` instead